doc/specs/#1595 - Suggestions UI/Snippets.md

```diff
@@ -0,0 +1,526 @@
+* [ ] When the user `cd`s to a directory (with shell integration enabled), the
+  terminal can load the tasks from that directory tree
+* [ ] [#10436] Users can manage all their fragments extensions directly in the Settings UI
+* [ ] The suggestions UI & snippets pane can filter not only on name of a
+  command, but for snippets, the input as well.
+* [ ] [#12927] Enlighten the suggestions UI to support (_a yet undeclared syntax
+  for_) snippets with prompt-able sections in them
+
+<!--
+
+These are perhaps too vague to be included in the official spec, but they are left here for posterity:
+
+### 🚀 Sprint
+
+* [ ] Fork of [#12927] - prompt-able sections can accept a command to dynamically populate options
+* [ ] Enlighten the suggestions UI to understand a yet undeclared syntax for snippets with prompt-able sections in them.
+* [ ] Stand up a dedicated repo for `terminal-snippets`
+  * [ ] Pull those down, or compile them in, or so -->
+
+## Conclusion
+
+Snippets are something that developers immediately understand the value of.
+After talking with users, everyone we talked with immediately understood the
+concept, and you could see the gears turning on ways to integrate this into
+their own workflows.
+
+### Future Considerations
+
+* We may want to add additional params to the `save` subcommand in the future,
+  to configure where the snippet is saved:
+  * `--local`: Save to the `.wt.json` in the CWD, if there is one (or create
+    one)
+  * `--parent`: Save to the `.wt.json` in the first ancestor of the CWD, if
+    there is one. Otherwise create one here.
+  * `--settings`: Manually save to the settings file?
+  * `--profile`: save to this profile???? Not sure if this is actually possible.
+    Maybe with the `WT_SESSION_ID` env var to figure out which profile is in use
+    for the pane with that ID
+* Longer workflows might be better exposed as notebooks. We've already got a
+  mind to support [markdown in a notebook-like
+  experience](https://github.com/microsoft/terminal/issues/16495) in the
+  Terminal. For longer scripts that may need rich markup between commands, that
+  will likely be a better UX.
+* For what it is worth, [Warp] uses .yaml files for their "workflows". As an
+  example, see
+  [`clone_all_repos_in_org.yaml`](https://github.com/warpdotdev/workflows/blob/main/specs/git/clone_all_repos_in_org.yaml).
+  We may want to straight up just seamlessly support that syntax as well.
+  Converting them to WT-compatible json is fairly trivial [[1](#footnote-1)].
+  Furthermore, the commands are all licensed under Apache 2.0, which means they
+  can be easily consumed by other OSS projects and shared with other developers.
+  This leads us to the next future consideration:
+* Discoverability will be important. Perhaps the actions page could have a
+  toggle to immediately filter to "snippets"? Which then also displays some text
+  like "Tip: save snippets directly from the commandline with
+  `wt save <the commandline>`".

```

Comment:
Will Snippets be in the Settings UI? To what degree? I feel like that's missing from this spec.

You mention it a little bit here. I'd like it if we could commit to some of it being in the Settings UI in the walk/run phase. I'm concerned about discoverability and having it in the SUI would help, at least a little bit.

Reply:
I mean, they're just actions fundamentally. They're in the actions editor just the same as any other action. When I talked with @PankajBhojwani about this, we agreed that a dedicated Snippets page is basically just a _second_ actions page.  Maybe we could add a button to filter that list specifically to snippets... but there's a lot of work to be done on that page in general. Didn't feel like it was necessary to call out.

Reply:
Agreed, I think we can come back to SUI discoverability once the actions page has had its overhaul! The idea of a dedicated snippets page feels messy to me since those send input actions would also be in the actions page